### DISCUSSION TITLE
How to use Right Prepromt for kube context?

### DISCUSSION THREAD

**Discussion by nela** [sindresorhus/pure#629] [General]
Firstly, thanks for a cool and simple theme!

This is not really an issue, just looking for advice. I am looking to fork pure and fit it to my own needs. In that regard I'd like to add a right prepromt to display the kube context, and also move the execution time to the right preprompt.  I've been playing around with `RPROMPT`, but that gives me the output in line with prompt. 

Also wondering if you have any recommendations regarding implementation of [zsh-kubectl-prompt](https://github.com/superbrothers/zsh-kubectl-prompt) asynchronously?

**Comment by nela** [comment]
I\ve come accross [this post](http://aperiodic.net/phil/prompt) and it seems as the appending padding and right prompt stuff to `preprompt_parts` is the way to go. I do find it a bit tricky to calculate the padding due to the async nature of  preprompt update. 
Also, after browsing issues I see that my questions is somewhat related to #460 and #448. In any case, some guidance would be appreciated.

**Reply by mafredri** [reply]
@nela I converted this into a discussion as it's better suited for questions. I'm not exactly sure what the end-result you're trying to achieve is (I know what information you want in `RPROMPT`, but why do you e.g. need padding?). As for how to make kube prompt asynchronous, I think the best example of that is the git functionality in the Pure source code. If there's something you don't understand of how that's structured, feel free to ask.

**Reply by polymorcodeus** [reply]
I'm going to wager a guess that he would like the RPROMPT on the first line rather than the second, thereby needing the "padding" to adjust/accommodate the terminal window variable width. 

I say this as I'm encountering the same issue/want :)

**Reply by anphn** [reply]
@polymorcodeus i want rprompt on the first line, too. You can do it?

**Reply by polymorcodeus** [reply]
I meant to come back to this but forgot - so thanks for the ping @anphn.  I do enough k8s work that I wanted my ns/context displayed on that first line and recently came across pure so wanted to make it work.

I didn't do the padding per se but I did get something working. I found a random stackoverflow? post where someone else was doing similar but different and suggested doing a print "hack" to line up/down for the RPROMPT. I have another plugin that is responsible for the ZSH_KUBECTL_PROMPT content. (I normally have it blank if not connected)

There was some mention that on occasion it doesn't redraw appropriately but I haven't had problems yet. YMMV

```
## snippet from my .zshrc
_lineup=$'\e[1A'
_linedown=$'\e[1B'
_kutl='%F{red}$ZSH_KUBECTL_PROMPT%f'
RPROMPT=%{${_lineup}%}${_kutl}%{${_linedown}%}
```

What it looks like in practice:
<img width="536" alt="image" src="https://github.com/user-attachments/assets/6df9c6f5-2da7-4474-9ae2-e546919bcf01" />
